cli/tools/run/mod.rs

```diff
@@ -219,3 +237,143 @@ pub async fn maybe_npm_install(factory: &CliFactory) -> Result<(), AnyError> {
   }
   Ok(())
 }
+
+pub async fn run_eszip(
+  flags: Arc<Flags>,
+  run_flags: RunFlags,
+) -> Result<i32, AnyError> {
+  // TODO(bartlomieju): actually I think it will also fail if there's an import
+  // map specified and bare specifier is used on the command line
+  let factory = CliFactory::from_flags(flags.clone());
+  let cli_options = factory.cli_options()?;
+  let import_map_specifier =
+    cli_options.resolve_specified_import_map_specifier()?;
+
+  // entrypoint#path1,path2,...
+  let (entrypoint, files) = run_flags
+    .script
+    .split_once("#")
+    .with_context(|| "eszip: invalid script string")?;
+
+  println!("running eszip: entrypoint={} files={}", entrypoint, files);
+
+  let module_loader =
+    create_eszip_module_loader(files, import_map_specifier).await?;
+  todo!()
+}
+
+struct EszipModuleLoader {
```

Comment:
Probably better to store fully resolved URLs here, but need to have the "base" URL for eszip - I guess we could use CWD as a base for ESZip specifiers?